### PR TITLE
Internal: Revert PHP Dependency Injection [ED-15428]

### PR DESCRIPTION
**Pull request by Batsirai-muchareva**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

**Review by TzviRabinovitch — DISMISSED** [review]

File: tests/phpunit/elementor/includes/container/test-container.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+use Elementor\Core\Utils\Version;
+use Elementor\Plugin;
+use Elementor\Tests\Phpunit\Includes\Container\Traits\Trait_Test_Container;
+use ElementorDeps\DI\DependencyException;
+use ElementorDeps\DI\NotFoundException;
+use ElementorEditorTesting\Elementor_Test_Base;
+use ElementorDeps\DI as DI;
+
+class Test_Container extends Elementor_Test_Base
+{
```
Comment (louiswol94) [inline tests/phpunit/elementor/includes/container/test-container.php:18]:
```suggestion
class Test_Container extends Elementor_Test_Base {
```
Check the code styling of this file

File: tests/playwright/upgrade-test/index.ts [hunk 2]
```diff
@@ -24,7 +24,8 @@ export class UpgradeElementor {
 		// Don't forget to npx grunt build in root folder for local testing
 		this.cmd( 'mkdir plugin' );
 		this.cmd( 'cd ../../../ && mkdir elementor' );
-		this.cmd( 'cd ../../../ && cp -r ./build/* elementor && zip -r elementor elementor' );
+		this.cmd( 'cd ../../../ && cp -r ./build/* elementor' );
+		this.cmd( 'cd ../../../elementor && zip -r ../elementor.zip .' );
```
Comment (louiswol94) [inline tests/playwright/upgrade-test/index.ts:28]: Is this necessary?

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/includes/container/traits/trait-test-container.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Elementor\Tests\Phpunit\Includes\Container\Traits;
+
+trait Trait_Test_Container
+{
```
Comment (louiswol94) [inline tests/phpunit/elementor/includes/container/traits/trait-test-container.php:6]:
```suggestion
trait Trait_Test_Container {
```
This file as well

File: php-scoper/dependency-injection.inc.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
```
Comment (louiswol94) [inline php-scoper/dependency-injection.inc.php:3]:
```suggestion
declare( strict_types = 1 );
```
Do we need it?

File: includes/container/container.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	public static function get_instance() {
+		if ( is_null( static::$instance ) ) {
+			self::initialize_instance();
+		}
+
+		return static::$instance;
+	}
+}
+
+
```
Comment (louiswol94) [inline includes/container/container.php:59]:
```suggestion

```

File: .gitignore [hunk 6]
```diff
@@ -40,3 +41,4 @@ tests/playwright/blob-report
 **/allure-results
 playwright-test-results
 **/.env
+
```
Comment (louiswol94) [inline .gitignore:44]:
```suggestion
```

**Review by louiswol94 — DISMISSED** [review]

File: elementor.php [hunk 7]
```diff
@@ -43,6 +43,13 @@
 define( 'ELEMENTOR_ASSETS_PATH', ELEMENTOR_PATH . 'assets/' );
 define( 'ELEMENTOR_ASSETS_URL', ELEMENTOR_URL . 'assets/' );
 
+if ( file_exists( ELEMENTOR_PATH . 'vendor/autoload.php' ) ) {
+	require_once ELEMENTOR_PATH . 'vendor/autoload.php';
+	// we need this file because of the DI\create function we are using
```
Comment (hein-obox) [inline elementor.php:48]:
```suggestion
	// We need this file because of the DI\create function that we are using.
```

File: elementor.php [hunk 8]
```diff
@@ -43,6 +43,13 @@
...
+	// autoload classmap doesn't include this file
```
Comment (hein-obox) [inline elementor.php:49]:
```suggestion
	// Autoload classmap doesn't include this file.
```

File: includes/container/container.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+use ElementorDeps\DI\Container as DIContainer;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Elementor Container.
+ *
+ * Elementor container handler class is responsible for containerization
```
Comment (hein-obox) [inline includes/container/container.php:15]:
```suggestion
 * Elementor container handler class is responsible for the containerization
```

File: includes/container/container.php [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+ * manager classes and their dependencies.
```
Comment (hein-obox) [inline includes/container/container.php:16]:
```suggestion
 * of manager classes and their dependencies.
```

File: tests/phpunit/elementor/includes/container/traits/trait-test-container.php [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Elementor\Tests\Phpunit\Includes\Container\Traits;
+
+trait Trait_Test_Container {
+	public function bind($abstract, $concrete): void
```
Comment (hein-obox) [inline tests/phpunit/elementor/includes/container/traits/trait-test-container.php:6]:
```suggestion
	public function bind($abstract, $concrete): void {
```

File: tests/phpunit/elementor/includes/container/traits/trait-test-container.php [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+	{
```
Comment (hein-obox) [inline tests/phpunit/elementor/includes/container/traits/trait-test-container.php:7]:
```suggestion
```

File: tests/phpunit/elementor/includes/container/test-container.php [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+		$versionClass = $this->container->get( Version::class );
+
+		$this->assertEquals( $major_version1, $versionClass->major1 );
+		$this->assertEquals( $major_version2, $versionClass->major2 );
+		$this->assertEquals( $patch, $versionClass->patch );
+		$this->assertEquals( $stage, $versionClass->stage );
+	}
+
+	public function test_alias_returns_same_Instance() {
```
Comment (hein-obox) [inline tests/phpunit/elementor/includes/container/test-container.php:94]: Maybe?

```suggestion
	public function test_alias_returns_same_instance() {
```

**Review by hein-obox — APPROVED** [review]

**Pull request merged**